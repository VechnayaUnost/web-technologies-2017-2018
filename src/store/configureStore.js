import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware, { END } from 'redux-saga';

import rootReducer from '../reducers';
import rootSaga from '../sagas';

export default (initialState) => {
    const sagaMiddleware = createSagaMiddleware();
    const store = createStore(rootReducer, initialState, applyMiddleware(sagaMiddleware));

    sagaMiddleware.run(rootSaga);
    store.runSaga = () => sagaMiddleware.run(rootSaga);
    store.close = () => store.dispatch(END);

    return store;
};