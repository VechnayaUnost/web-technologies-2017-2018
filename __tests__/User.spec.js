import React from 'react';
import User from '../src/components/User';
import { shallow, mount, configure } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

configure({ adapter: new Adapter() });

describe('User component test', () => {
    const mockStore = configureStore([ thunk ]);
    const initialState = {
        userInfo: {
            avatar_url: 'https://vignette.wikia.nocookie.net/james-camerons-avatar/images/2/2c/Neytiri_Avatar.jpg/revision/latest/scale-to-width-down/640?cb=20140428095608&path-prefix=ru',
            name: 'Neytiri',
            login: 'GoGoPandora'
        }
    }

    let store;

    beforeEach(() => {
        store = mockStore(initialState);
    })

    it('User component rendered succesfully', () => {
        const user = mount(<User store={store} />);
        expect(mountToJson(user)).toMatchSnapshot();
    });

    it('User props match store', () => {
        const user = shallow(<User store={store}/>);
        expect(user.prop('name')).toBe('Neytiri');
        expect(user.prop('login')).toBe('GoGoPandora');
    })
});