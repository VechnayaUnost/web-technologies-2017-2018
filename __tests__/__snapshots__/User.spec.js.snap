// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component test User component rendered succesfully 1`] = `
<Connect(User)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <User
    avatar_url="https://vignette.wikia.nocookie.net/james-camerons-avatar/images/2/2c/Neytiri_Avatar.jpg/revision/latest/scale-to-width-down/640?cb=20140428095608&path-prefix=ru"
    dispatch={[Function]}
    login="GoGoPandora"
    name="Neytiri"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div
      className="User"
    >
      <img
        alt="Neytiri"
        src="https://vignette.wikia.nocookie.net/james-camerons-avatar/images/2/2c/Neytiri_Avatar.jpg/revision/latest/scale-to-width-down/640?cb=20140428095608&path-prefix=ru"
      />
      <h2>
        Neytiri
      </h2>
      <h3>
        GoGoPandora
      </h3>
    </div>
  </User>
</Connect(User)>
`;
